### PR TITLE
Extract messaging and editor

### PR DESCRIPTION
**Pull request by rgbkrk**
Extracted the editor and the messaging.

The only part of the editor that is extracted is our base editor component and the view version (@peggyrayzis can explain better than I).

### CODE REVIEW

**Comment by rgbkrk** [comment]
I'm going to leave the messaging/session refactor for another day as it would make this PR messier than intended. Mostly just want the editor component out.

File: packages/editor/__tests__/editor-spec.js [hunk 1]
```diff
@@ -2,19 +2,14 @@ import React from 'react';
...
+import WrappedEditor from '../src/view';
 
-chai.use(sinonChai);
-
-const complete = require('../../../../src/notebook/components/cell/codemirror/complete');
+const complete = require('../src/complete');
 
 describe('WrappedEditor', () => {
-  it('reaches out for code completion', (done) => {
+  it.skip('reaches out for code completion', (done) => {
```
Comment (rgbkrk) [inline packages/editor/__tests__/editor-spec.js:12]: I had trouble mocking one function from `complete` here so I've skipped this test for now.

**Comment by rgbkrk** [comment]
Once this is in, I'll bring it into the notebook preview PR.

File: packages/editor/__tests__/editor-spec.js [hunk 2]
```diff
@@ -2,19 +2,14 @@ import React from 'react';
 import Rx from 'rxjs/Rx';
 
 import { mount } from 'enzyme';
-import chai, { expect } from 'chai';
```
Comment (jdetle) [inline packages/editor/__tests__/editor-spec.js:5]: Where is expect getting imported from now? Or is it no longer needed in jest?

**Review by jdetle — APPROVED** [review]

File: packages/editor/__tests__/editor-spec.js (hunk 2 above)
Comment (jdetle) [reply packages/editor/__tests__/editor-spec.js:5]: Ahh, it appears this is a benefit of Jest, awesome!

**Comment by peggyrayzis** [comment]
Just getting around to looking at this, it looks great! Thanks for extracting the editor package 😀

**Pull request merged**